ui/app/search/search-ctrl.js

```diff
@@ -87,10 +79,11 @@
         mlSearch.addAdditionalQuery(combinedQuery);
         search();
       }
-
+      
       function search(qtext) {
         if ( arguments.length ) {
           model.qtext = qtext;
+          $location.search().q = qtext;
```

Comment:
The url that may contain a q will force the search later on to make use of that old value. The alternative to this is to modify the directive's controller or override the directive's behavior.

However, that change would require various wiring that imo, introduces too much complexity relative to the goal.